kimchi/src/verifier_index.rs

```diff
@@ -236,6 +238,24 @@ impl<G: CommitmentCurve> VerifierIndex<G>
 where
     G::BaseField: PrimeField,
 {
+    pub fn srs(&self) -> &Arc<SRS<G>> {
+        self.srs.get_or_init(|| {
+            let mut srs = SRS::<G>::create(self.max_poly_size);
```

Comment:
I don't think we want to create a new SRS here, for memory usage / performance reasons. Can we continue to use the one from the prover index?

Reply:
I think you should change the field to be `Option<Arc<SRS>>`, like that it will deserialize to `None`. You can then implement `srs()` as:

```rust
fn srs(&self) -> Arc<SRS> {
  self.srs.expect("You must use `set_srs` to initialize the SRS")
}

  fn set_srs(&mut self, srs: Arc<SRS>) {
    self.srs = Some(srs)
  }
```

Reply:
but that loses the ‘auto’ deserialization property though. or we dont really need that for srs?

Reply:
Oh no you’re right. I was thinking that we’d never want to do this, but actually we’d probably want to do it. We can probably use max_poly_size in the verifier index to init it if it’s not initialized